packages/docusaurus-theme-classic/src/theme/Navbar/ColorModeToggle/index.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useColorMode, useThemeConfig} from '@docusaurus/theme-common';
+import ColorModeToggle from '@theme/ColorModeToggle';
+import type {Props} from '@theme/Navbar/ColorModeToggle';
+import React from 'react';
+
+export default function NavbarColorModeToggle({
+  className,
+}: Props): JSX.Element | null {
```

Comment:
I wonder if this layer of abstraction is useful? Supposing the entire purpose of `ColorModeToggle` is to toggle color modes, should we just move the hook calls to `ColorModeToggle` instead and remove this component?

Reply:
There is value to have design system components (ColorModeToggle) to be very simple and have limited dependencies (ie not using any context). This makes it much easier to test then in isolation, add them to Storybook and other things that I plan to do.

This abstraction is likely to be temporary anyway if we provide a NavbarColorModeItem. For now I assume it's good enough, as this component is not marked as safe anyway

The general idea is that each call site should also be responsible to integrate the toggle on a case-by-case basis. For example, rendering null in case toggle is disabled might not be the best solution in all cases: it really depends on the parent component layout.

If we have later an option to have a sidebar color mode toggle, we'd create a similar SidebarColorModeToggle component, allowing the user to easily have distinct toggle designs for navbar/sidebar

<img width="329" alt="image" src="https://user-images.githubusercontent.com/749374/158631315-c607b430-94cc-4a45-a3ae-abe87692cb6f.png">

Reply:
I see, makes sense